tests/pipelines/stable_diffusion_3/test_onnx_pipeline_stable_diffusion_3.py

```diff
@@ -0,0 +1,297 @@
+            torch.randn(2, pooled_projection_dim).to(device=torch_device, dtype=dtype),
+            torch.randn(2).to(device=torch_device, dtype=dtype),
+        ),
+        output_path=transformer_path,
+        ordered_input_names=["hidden_states", "encoder_hidden_states", "pooled_projections", "timestep"],
+        output_names=["out_sample"],  # has to be different from "sample" for correct tracing
+        dynamic_axes={
+            "hidden_states": {0: "batch", 1: "channels", 2: "height", 3: "width"},
+            "encoder_hidden_states": {0: "batch", 1: "sequence", 2: "embed_dims"},
+            "pooled_projections": {0: "batch", 1: "projection_dim"},
+            "timestep": {0: "batch"},
+        },
+    )
+
+    # VAE ENCODER
+    vae_encoder = pipeline.vae
+    vae_in_channels = vae_encoder.config.in_channels
+    vae_sample_size = vae_encoder.config.sample_size
+    vae_encoder.forward = lambda sample, return_dict: vae_encoder.encode(sample, return_dict)[0].sample()
+    onnx_export(
+        vae_encoder,
+        model_args=(
+            torch.randn(1, vae_in_channels, vae_sample_size, vae_sample_size).to(device=torch_device, dtype=dtype),
+            False,
+        ),
+        output_path=Path("sd3/vae_encoder/model.onnx"),
+        ordered_input_names=["sample", "return_dict"],
+        output_names=["latent_sample"],
+        dynamic_axes={
+            "sample": {0: "batch", 1: "channels", 2: "height", 3: "width"},
+        },
+    )
+
+    # VAE DECODER
+    vae_decoder = pipeline.vae
+    vae_latent_channels = vae_decoder.config.latent_channels
+    vae_out_channels = vae_decoder.config.out_channels
+    vae_decoder.forward = vae_encoder.decode
+    onnx_export(
+        vae_decoder,
+        model_args=(
+            torch.randn(1, vae_latent_channels, sample_size, sample_size).to(device=torch_device, dtype=dtype),
+            False,
+        ),
+        output_path=Path("sd3/vae_decoder/model.onnx"),
+        ordered_input_names=["latent_sample", "return_dict"],
+        output_names=["sample"],
+        dynamic_axes={
+            "latent_sample": {0: "batch", 1: "channels", 2: "height", 3: "width"},
+        },
+    )
+    del pipeline.vae
+
+    onnx_pipeline = OnnxStableDiffusion3Pipeline(
+        vae_encoder=OnnxRuntimeModel.from_pretrained("sd3/vae_encoder"),
+        vae_decoder=OnnxRuntimeModel.from_pretrained("sd3/vae_decoder"),
+        text_encoder=OnnxRuntimeModel.from_pretrained("sd3/text_encoder"),
+        tokenizer=pipeline.tokenizer,
+        text_encoder_2=OnnxRuntimeModel.from_pretrained("sd3/text_encoder_2"),
+        tokenizer_2=pipeline.tokenizer_2,
+        text_encoder_3=OnnxRuntimeModel.from_pretrained("sd3/text_encoder_3"),
+        tokenizer_3=pipeline.tokenizer_3,
+        transformer=OnnxRuntimeModel.from_pretrained("sd3/transformer"),
+        scheduler=pipeline.scheduler,
+    )
+
+    onnx_pipeline.save_pretrained("sd3")
```

Comment:
Since I have no access to upload ONNX SD3 model to hf-internal-testing, I export the ONNX model in UT. If anyone could help upload it, the code can be simplified.